quisp/modules/QRSA/HardwareMonitor/HardwareMonitor.cc

```diff
@@ -511,6 +577,145 @@ Matrix4cd HardwareMonitor::reconstruct_Density_Matrix(int qnic_id) {
   density_matrix_Y2.real() ).trace(); EV<<"Yerr = "<<Yerr_rate2<<"\n";*/
 }
 
+Matrix4cd HardwareMonitor::extended_reconstruct_density_matrix(int qnic_id, int partner) {
```

Comment:
we should separate this method from this class and add a unit test for this later.